qiskit/transpiler/passes/scheduling/base_scheduler.py

```diff
@@ -261,10 +261,21 @@ def _get_node_duration(
     ) -> int:
         """A helper method to get duration from node or calibration."""
         indices = [dag.find_bit(qarg).index for qarg in node.qargs]
+        is_mid_circuit = any(
+            dag.descendants(successor) != set() for successor in dag.successors(node)
+        )
```

Comment:
It might be more efficient to check this as:

```suggestion
        is_mid_circuit = not any(
            isinstance(x, DAGOutNodes) for x in dag.quantum_successors(node) 
        )
```
If the measurement is the last operation on qubit it's immediate successor will be of type `DAGOutNode` so we can just iterate over the the quantum successors of a node and determine if it's at the end or not.